data-prepper-core/src/main/java/org/opensearch/dataprepper/parser/PipelineParser.java

```diff
@@ -130,7 +134,8 @@ private void buildPipelineFromConfiguration(
                     .map(this::buildSinkOrConnector)
                     .collect(Collectors.toList());
 
-            final Pipeline pipeline = new Pipeline(pipelineName, source, buffer, decoratedProcessorSets, sinks, processorThreads, readBatchDelay);
+            final long processorShutdownTimeout = dataPrepperConfiguration.getProcessorShutdownTimeout().toMillis();
```

Comment:
Let's use the `Duration` as far as possible. No need to convert to millis until it is fully needed.